src/main/java/com/tabnine/binary/BinaryProcessRequesterProvider.java

```diff
@@ -105,7 +105,12 @@ public void onTimeout() {
 
     Logger.getInstance(getClass()).info("TabNine's response timed out.");
 
-    if (++consecutiveTimeouts >= timeoutsThreshold) {
+    long now = System.currentTimeMillis();
+    if (firstTimeoutTimestamp == null) {
+      firstTimeoutTimestamp = now;
```

Comment:
You don't want to set it back to null on a good request?

Reply:
I do it